examples/rnn_classifer/RNN_sentiment_classification.ipynb

```diff
@@ -426,33 +428,47 @@
     "            num_units=128,\n",
     "            num_layers=2,\n",
     "            dropout=0,\n",
+    "            bidirectional=False,\n",
     "    ):\n",
     "        super().__init__()\n",
     "        self.embedding_dim = embedding_dim\n",
     "        self.rec_layer_type = rec_layer_type.lower()\n",
     "        self.num_units = num_units\n",
     "        self.num_layers = num_layers\n",
     "        self.dropout = dropout\n",
+    "        self.bidirectional = bidirectional\n",
     "\n",
+    "        self.reset_weights()\n",
+    "\n",
+    "    def reset_weights(self):\n",
     "        self.emb = nn.Embedding(VOCAB_SIZE + 1, embedding_dim=self.embedding_dim)\n",
     "        \n",
     "        rec_layer = {'lstm': nn.LSTM, 'gru': nn.GRU}[self.rec_layer_type]\n",
     "        # We have to make sure that the recurrent layer is batch_first,\n",
     "        # since sklearn assumes the batch dimension to be the first\n",
     "        self.rec = rec_layer(\n",
-    "            self.embedding_dim, self.num_units, num_layers=num_layers, batch_first=True)\n",
+    "            self.embedding_dim,\n",
+    "            self.num_units,\n",
+    "            num_layers=self.num_layers,\n",
+    "            dropout=self.dropout,\n",
+    "            bidirectional=self.bidirectional,\n",
+    "            batch_first=True,\n",
+    "        )\n",
     "\n",
+    "        self.drop = nn.Dropout(self.dropout)\n",
     "        self.output = nn.Linear(self.num_units, 2)\n",
     "\n",
     "    def forward(self, X):\n",
     "        embeddings = self.emb(X)\n",
+    "\n",
     "        # from the recurrent layer, only take the activities from the last sequence step\n",
     "        if self.rec_layer_type == 'gru':\n",
     "            _, rec_out = self.rec(embeddings)\n",
     "        else:\n",
     "            _, (rec_out, _) = self.rec(embeddings)\n",
     "        rec_out = rec_out[-1]  # take output of last RNN layer\n",
-    "        drop = F.dropout(rec_out, p=self.dropout)\n",
```

Comment:
I am guessing this was a bug because we are not able to set the `training` kwarg parameter in the functional interface?

Reply:
Exactly, as is, we always apply dropout, even at inference time.